sections/Deals/AcceptOrReject/AcceptOrRejectSection/AcceptOrRejectSection.tsx

```diff
@@ -452,84 +466,90 @@ const AcceptOrRejectDeal: FC<AcceptOrRejectDealProps> = ({
 		async function getGasLimitEstimate() {
 			if (!walletAddress || !signer || !deal?.poolAddress || !poolBalances?.purchaseTokenDecimals)
 				return setGasLimitEstimate(null);
+
 			const contract = new ethers.Contract(deal.poolAddress, poolAbi, signer);
-			if (txType === TransactionType.Withdraw && isMaxValue) {
-				setGasLimitEstimate(wei(await contract.estimateGas.withdrawMaxFromPool(), 0));
-			} else if (txType === TransactionType.Withdraw) {
-				setGasLimitEstimate(
-					wei(
-						await contract.estimateGas.withdrawFromPool(
-							ethers.utils.parseUnits(
-								(inputValue ?? 0).toString(),
-								poolBalances?.purchaseTokenDecimals ?? 0
-							)
-						),
-						0
-					)
-				);
-			} else if (txType === TransactionType.Accept && isMaxValue) {
-				setGasLimitEstimate(wei(await contract.estimateGas.acceptMaxDealTokens(), 0));
-			} else if (txType === TransactionType.Accept) {
-				setGasLimitEstimate(
-					wei(
-						await contract.estimateGas.acceptDealTokens(
-							ethers.utils.parseUnits(
-								(inputValue ?? 0).toString(),
-								poolBalances?.purchaseTokenDecimals ?? 0
-							)
-						),
-						0
-					)
-				);
-			} else {
-				return setGasLimitEstimate(null);
+
+			const txTypeOptions = {
+				[TransactionType.Withdraw]: async () => {
+					if (isMaxValue) {
+						return setGasLimitEstimate(wei(await contract.estimateGas.withdrawMaxFromPool(), 0));
+					}
+
+					return setGasLimitEstimate(
+						wei(
+							await contract.estimateGas.withdrawFromPool(
+								ethers.utils.parseUnits(
+									(isEmptyInput ? 0 : inputValue).toString(),
+									poolBalances?.purchaseTokenDecimals ?? 0
+								)
+							),
+							0
+						)
+					);
+				},
+				[TransactionType.Accept]: async () => {
+					if (isMaxValue) {
+						return setGasLimitEstimate(wei(await contract.estimateGas.acceptMaxDealTokens(), 0));
+					}
+
+					return setGasLimitEstimate(
+						wei(
+							await contract.estimateGas.acceptDealTokens(
+								ethers.utils.parseUnits(
+									(isEmptyInput ? 0 : inputValue).toString(),
+									poolBalances?.purchaseTokenDecimals ?? 0
+								)
+							),
+							0
+						)
+					);
+				},
+			};
+
+			// @ts-ignore
+			if (typeof txTypeOptions[txType] === 'function') {
```

Comment:
A bit overkill imo. No need for this. Just use the good old `if` statement as we used to do.

Reply:
Done